simpleclient_servlet/src/main/java/io/prometheus/client/filter/MetricsFilter.java

```diff
@@ -0,0 +1,80 @@
+package io.prometheus.client.filter;
+
+import io.prometheus.client.Counter;
+import io.prometheus.client.Histogram;
+import org.apache.commons.lang3.StringUtils;
+
+import javax.servlet.Filter;
+import javax.servlet.FilterChain;
+import javax.servlet.FilterConfig;
+import javax.servlet.ServletException;
+import javax.servlet.ServletRequest;
+import javax.servlet.ServletResponse;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+
+/**
+ * Created by andrewstuart on 11/19/16.
+ *
+ * The MetricsFilter class exists to provide a high-level filter that enables tunable collection of metrics for Servlet
+ * performance.
+ *
+ * By default, this filter will provide metrics that distinguish 3 levels deep for the request path
+ * (including servlet context path).
+ */
+public class MetricsFilter implements Filter {
+    public static final String PATH_COMPONENT_PARAM = "path-components";
+    public static final int DEFAULT_PATH_COMPONENTS = 3;
+    public static final String FILTER_NAME = "servlet_request_latency";
+
+    private static Histogram servletLatency = Histogram.build()
```

Comment:
You'll notice in my updated PR that this is still not final, but that it's now being initialized in the `init` method. That was my original intent, and forgot to finalize when I temporarily moved it back out of `init`.

Let me know if there's a better solution that I'm not thinking of.